### PR TITLE
DOC: Clarify difference between PySequence_GETITEM, PyArray_GETITEM 

### PR DESCRIPTION
**Pull request by yugkhanna**
~Closes  #10966.~

Added documentation for return of standard python types on success to avoid confusion between PySequence_GETITEM and PyArray_GETITEM

### CODE REVIEW

File: doc/source/reference/c-api.array.rst [hunk 1]
```diff
@@ -123,7 +123,8 @@ sub-types).
 .. c:function:: PyObject *PyArray_GETITEM(PyArrayObject* arr, void* itemptr)
 
     Get a Python object from the ndarray, *arr*, at the location
-    pointed to by itemptr. Return ``NULL`` on failure.
+    pointed to by itemptr. Return ``NULL`` on failure and return standard
+    Python types on success.
```
[the remark this answers is not in the document]
Comment (yugkhanna) [reply doc/source/reference/c-api.array.rst]: Okay. I'll do the changes

File: doc/source/reference/c-api.array.rst [hunk 2]
```diff
@@ -122,9 +122,11 @@ sub-types).
 
 .. c:function:: PyObject *PyArray_GETITEM(PyArrayObject* arr, void* itemptr)
 
-    Get a Python object from the ndarray, *arr*, at the location
-    pointed to by itemptr. Return ``NULL`` on failure.
-
+    Return (builtin) Python types and get a Python object from the ndarray, *arr*, 
+    at the location pointed to by itemptr. Return ``NULL`` on failure.
+    
+    `numpy.ndarray.item <https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.item.html>`_ is identical to PyArray_GETITEM.
```
Comment (eric-wieser) [inline doc/source/reference/c-api.array.rst]: No need for the `<...>` here - using `` `numpy.ndarray.item` `` is enough

File: doc/source/reference/c-api.array.rst [hunk 3]
```diff
@@ -122,9 +122,11 @@ sub-types).
 
 .. c:function:: PyObject *PyArray_GETITEM(PyArrayObject* arr, void* itemptr)
 
-    Get a Python object from the ndarray, *arr*, at the location
-    pointed to by itemptr. Return ``NULL`` on failure.
-
+    Return (builtin) Python types and get a Python object from the ndarray, *arr*,
```
Comment (eric-wieser) [inline doc/source/reference/c-api.array.rst]: How about `Get a python object of a builtin type from the array`?

Comment (yugkhanna) [reply doc/source/reference/c-api.array.rst]: Sounds good

**Review by eric-wieser — APPROVED** [review]

**Comment by yugkhanna** [comment]
@eric-wieser Okay sounds good. You can merge this as is and next I will add the differences to close that issue.

**Comment by eric-wieser** [comment]
In future, you should make a new branch for each PR, rather than working on `master`

**Pull request merged**